vmdb/app/models/ems_refresh/parsers/kubernetes.rb

```diff
@@ -76,9 +76,10 @@ def parse_pod(pod)
       end
       new_result.merge!(
         # namespace is overriden in more_core_extensions and hence needs a non method access
-        :namespace      => pod.metadata.instance_values["table"][:namespace],
-        :restart_policy => pod_restart_policy,
-        :dns_policy     => pod.spec.dnsPolicy
+        :namespace           => pod.metadata.instance_values["table"][:namespace],
+        :restart_policy      => pod_restart_policy,
+        :dns_policy          => pod.spec.dnsPolicy,
+        :container_node_name => (defined? pod.spec.host) ? pod.spec.host : nil
```

Comment:
I don't think `defined?` is what you want here.  What are you trying to do...see if the host key is there?